### PR TITLE
Adds a sealant gun.

### PR DESCRIPTION
**Pull request by MistakeNot4892**
- Adds a sealant gun which is absolutely not the GLOO gun from Prey because it's named differently.
- The sealant gun uses a tank of pressurized metal foam to fire globs and make metal walls from a distance.
- It can be drag-fired or single-shot.
- If you shoot a human they will get gooped up and will be slowed down until they remove the sealant.

**TODO:**
- Find someone to make sprites for the sealant gun, sealant tank rack and sealant tank injector.
- Add a UI to the sealant injector so cartridges can be replaced/configured properly.

### CODE REVIEW

File: code/modules/sealant_gun/sealant.dm [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+	color = "#cccdcc"
+	slowdown_general = 3
+	tint = TINT_BLIND
+	canremove = FALSE
+	slot_flags = SLOT_HEAD | SLOT_MASK | SLOT_OCLOTHING | SLOT_FEET
+
+	var/foam_type = /obj/structure/foamedmetal
+	var/splatted = FALSE // Used to prevent doubled effects if throwcode wonks up.
+	var/hardened = FALSE // Set manually post-equip so that the blob can't be removed without breaking.
+	var/global/list/splat_try_equip_slots = list(
```
Comment (Lohikar) [inline code/modules/sealant_gun/sealant.dm:18]: nit: I feel like `static` is clearer than `global` for instance statics.

File: code/modules/sealant_gun/sealant.dm [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+
+/obj/item/clothing/sealant/dropped(mob/user)
+	. = ..()
+	break_apart()
+
+/obj/item/clothing/sealant/proc/break_apart(var/mob/user)
+	canremove = TRUE
+	if(user)
+		user.unEquip(src, user.loc)
+		user.visible_message(SPAN_NOTICE("\The [user] smashes \the [src]!"))
```
Comment (Lohikar) [inline code/modules/sealant_gun/sealant.dm:51]: Maybe should add a self & blind message here.

File: code/modules/sealant_gun/sealant_gun.dm [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+	release_force = 5
+	fire_delay =    1
+
+	var/foam_charges_per_shot = 1
+	var/obj/item/sealant_tank/loaded_tank
+
+/obj/item/gun/launcher/sealant/on_update_icon()
+	cut_overlays()
+	if(loaded_tank)
+		add_overlay(image(icon, "tank"))
```
Comment (Lohikar) [inline code/modules/sealant_gun/sealant_gun.dm:24]:
```suggestion
		add_overlay("tank")
```

File: code/modules/sealant_gun/sealant_rack.dm [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+			LAZYADD(tanks, O)
+			update_icon()
+			return TRUE
+
+	. = ..()
+
+/obj/structure/sealant_rack/Initialize(ml, _mat, _reinf_mat)
+	. = ..()
+	for(var/i = 1 to rand(1,max_tanks))
+		LAZYADD(tanks, new /obj/item/sealant_tank(src))
```
Comment (Lohikar) [inline code/modules/sealant_gun/sealant_rack.dm:60]: Imo, cleaner to do:
```dm
LAZYINITLIST(tanks)
for(var/i = 1 to rand(1,max_tanks))
    tanks += new /obj/item/sealant_tank(src)
```

File: code/modules/sealant_gun/sealant_tank.dm [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+	name = "sealant tank"
+	desc = "A sealed tank used to keep hull sealant foam contained under pressure."
+	icon = 'icons/obj/sealant_tank.dmi'
+	icon_state = "tank"
+	var/foam_charges = 0
+	var/max_foam_charges = 60
+
+/obj/item/sealant_tank/on_update_icon()
+	cut_overlays()
+	add_overlay(image(icon, "fill_[Floor((foam_charges/max_foam_charges) * 5)]"))
```
Comment (Lohikar) [inline code/modules/sealant_gun/sealant_tank.dm:11]:
```suggestion
	add_overlay("fill_[Floor((foam_charges/max_foam_charges) * 5)]")
```

**Comment by MistakeNot4892** [comment]
Closing my draft PRs for now to clean up the PR list.

**Pull request closed without merging**